File: nativeQt/test/PathPainter/PathPainter.cpp
#include "PathPainter.h"
#include <QPainter>

PathPainter::PathPainter(QQuickItem *parent)
    : QQuickPaintedItem(parent)
{
}
void PathPainter::paint(QPainter *painter)
{
    painter->setPen(Qt::yellow);
    painter->setRenderHints(QPainter::Antialiasing, true);
    painter->drawLine(0, 0, 100, 100);
}

File: AbandonedGolangVersion/MainPrj/PathPainter.go
package main

import (
	"github.com/therecipe/qt/gui"
	"github.com/therecipe/qt/quick"
)

func init() {
	PathPainter_QmlRegisterType2("CustomViews", 1, 0, "PathPainter")
}

type PathPainter struct {
	quick.QQuickPaintedItem

	_ func() `constructor:"init"`

	_ string      `property:"name"`
	_ *gui.QColor `property:"color"`
}

func (p *PathPainter) init() {
	p.ConnectPaint(p.paint)
}

func (p *PathPainter) paint(painter *gui.QPainter) {
	
}